sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -163,6 +163,51 @@ final class CastExprMeta[INPUT <: UnaryExpression with TimeZoneAwareExpression w
   override protected val needTimezoneTagging: Boolean = false
 }
 
+trait CastOptions extends Serializable {
+  // The brackets that are used in casting structs and maps to strings
+  def leftBracket: String
+
+  def rightBracket: String
+
+  // The string value to use to represent null elements in array/struct/map.
+  def nullString: String
+
+  def useDecimalPlainString: Boolean
+
+  def useHexFormatForBinary: Boolean
+
+  def stringToDateAnsiModeEnabled: Boolean
+
+  def ansiMode: Boolean
+
+  def legacyCastComplexTypesToString: Boolean
```

Comment:
nit: this is kind of complicated because it appears to insert a space before a comma in some cases and it is not really clear what this does on its own separated from leftBracket, rightBracket, and nullString.